core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -773,6 +773,33 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
       }
     }
 
+  def memoize: ZManaged[R, E, ZManaged[R, E, A]] =
+    ZManaged {
+      RefM.make[Option[(Reservation[R, E, A], Exit[E, A])]](None).map { ref =>
+        val acquire1: ZIO[R, E, A] =
+          ref.modify {
+            case v @ Some((_, e)) => ZIO.succeed(e -> v)
+            case None =>
+              ZIO.uninterruptibleMask { restore =>
+                self.reserve.flatMap { res =>
+                  restore(res.acquire).run.map(e => e -> Some(res -> e))
+                }
+              }
+          }.flatMap(ZIO.done)
+
+        val release1 = (_: Exit[_, _]) => ZIO.unit
+
+        val acquire2: ZIO[R, E, ZManaged[R, E, A]] =
+          ZIO.succeed(ZManaged(ZIO.succeed(Reservation(acquire1, release1))))
```

Comment:
```suggestion
          ZIO.succeed(acquire1.toManaged_)
```

Reply:
:-) this looks better indeed - thanks for spotting